state-transition/core/types.go

```diff
@@ -33,32 +33,6 @@ import (
 	"github.com/karalabe/ssz"
 )
 
-// Context defines an interface for managing state transition context.
-type Context interface {
```

Comment:
I think this Context interface should still be used so that only the read-only methods are usable in this package. Currently the Setters can still be called from within state-transition

Reply:
Added a `transition.ReadOnlyContext` interface